taichi/backends/cuda/codegen_cuda.cpp

```diff
@@ -105,41 +105,40 @@ class CodeGenLLVMCUDA : public CodeGenLLVM {
   }
 
   void visit(PrintStmt *stmt) override {
+    // TODO(@archibate): ask @yuanming-hu about how to deal with CUDA print
+
     TI_ASSERT(stmt->width() == 1);
 
-    auto value_type = tlctx->get_data_type(stmt->stmt->ret_type.data_type);
+    std::vector<llvm::Type *> types;
+    std::vector<llvm::Value *> values;
 
-    std::string format;
+    std::string formats;
+    for (auto &&content : stmt->contents) {
+      if (formats.size() != 0)  // not the first expr?
+        formats += " ";         // add seperator
```

Comment:
Just FYI, there's a `fmt::join()` API, see https://fmt.dev/latest/api.html?highlight=join#_CPPv2N3fmt4joinERK5Range11string_view. But I'm fine with this approach..

Reply:
Cool! Btw, does C++ have something like the Python `map()`?

Reply:
Yes, and i thought you just used... `std::transform`

Reply:
I see, but can I use it as a return value? e.g. `std::vector<T> transform(std::vector<T>)`?

Reply:
```cpp
std::vector<Foo> in;
std::vector<Bar> out;
// Make sure out.size() >= in.size();
std::transform(in.begin(), in.end(), out.begin(), [](const Foo&)->Bar {...});
```